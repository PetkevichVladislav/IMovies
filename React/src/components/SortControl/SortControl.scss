.sort-control {
    color: #FFFFFF;
    display: flex;
    align-items: center;
    text-transform: uppercase;
    font-size: 20px;

    &__text {
        margin: 0px;

        &--sorting {
            color: #a8a8a8;
            margin: 0px 13px 0px 30px;
        }
    }

    &__select-container {
        position: relative;
        align-items: center;
    }

    &__select {
        border: none;
        background-color: transparent;
        color: #FFFFFF;
        font-size: 20px;
        appearance: none;
        padding-right: 25px;

        &:focus {
            outline: none;
            border: none;
        }

        option {
            background-color: #232323;
            font-size: 20px;
            border: none;
            &:hover {
                box-shadow: 0 0 10px 100px #F65261 inset;
            }
        }
    }
}

.icon-arrow-down {
    position: absolute;
    right: 17px;
    top: -3px;

    &::before {
        content: "\e900";
        display: inline-block;
        width: 0;
        height: 0;
        color: #F65261;
        font-size: 7px;
    }
}