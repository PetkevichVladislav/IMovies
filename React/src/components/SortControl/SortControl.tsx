import "./SortControl.scss"

import { FC, useCallback } from "react";
import { SortOption } from "../../models/enum/SortOption";
import { useMovieContext } from "../MovieListPage/MovieListPage";

export interface ISortControl {
    sortingOption: SortOption,    
}

export const SortControl: FC<ISortControl> = ({ sortingOption }) => {
    const movieContextModel = useMovieContext();    
    const handleSortingOptionChange = useCallback((event: React.ChangeEvent<HTMLSelectElement>) => {
        movieContextModel?.setSortCriterion(event.target.value as SortOption);
    }, [movieContextModel]);

    return (
        <div className="sort-control">
            <p className="sort-control__text sort-control__text--sorting">SORT BY</p>
            <div className="sort-control__select-container">
                <select className="sort-control__select" value={movieContextModel?.sortCriterion} data-testid="selected-sort-option" onChange={handleSortingOptionChange}>
                    {Object.values(SortOption).map((sortOption) => (
                        <option className="sort-control__text sort-control__text--sorting-option" key={sortOption} value={sortOption}>
                            {sortOption}
                        </option>
                    ))}
                </select>
                <span className="icon-arrow-down"></span>
            </div>
        </div>
    );
}