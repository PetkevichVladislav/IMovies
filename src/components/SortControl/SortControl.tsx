import { FC, useState } from "react";
import "./SortControl.scss"

export enum SortOption {
    ReleaseDate = "RELEASE DATE",
    Title = "TITLE",
}

export interface ISortControl {
    sortingOption: SortOption;
    onChange: (sortOption: SortOption) => void,
}

export const SortControl: FC<ISortControl> = ({ sortingOption, onChange }) => {
    const [selectedSortOption, setSelectedSortOption] = useState(sortingOption);

    function handleSortingOptionChange(event: React.ChangeEvent<HTMLSelectElement>) {
        setSelectedSortOption(event.target.value as SortOption);
    }
    return (
        <div className="sort-control">
            <p className="sort-control__text sort-control__text--sorting">SORT BY</p>
            <div className="sort-control__select-container">
                <select className="sort-control__select" value={selectedSortOption} data-testid="selected-sort-option" onChange={handleSortingOptionChange}>
                    {Object.values(SortOption).map((sortOption) => (
                        <option className="sort-control__text sort-control__text--sorting-option" key={sortOption} value={sortOption}>
                            {sortOption}
                        </option>
                    ))}
                </select>
                <span className="icon-arrow-down"></span>
            </div>
        </div>
    );
}