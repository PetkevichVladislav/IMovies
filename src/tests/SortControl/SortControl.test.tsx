import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { SortControl, SortOption } from '../../components/SortControl/SortControl';

describe("Sort control", () => {
    const onClickCallback = jest.fn();
    const sortControlData = {
        sortingOption : SortOption.ReleaseDate,
        onChange: (option: SortOption) => onClickCallback(option),
    };

    it("When data is valid component should renders correctly", () =>{
        const tree = renderer.create(<SortControl {...sortControlData}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();;
    });

    it("when user opens form options should be according to the enum", () =>{
        render(<SortControl {...sortControlData}/>);
        expect(screen.getByText(SortOption.ReleaseDate)).toBeInTheDocument();
        expect(screen.getByText(SortOption.Title)).toBeInTheDocument();
    });
});