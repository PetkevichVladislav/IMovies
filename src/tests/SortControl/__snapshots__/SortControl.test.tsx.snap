// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort control When data is valid component should renders correctly 1`] = `
<div
  className="sort-control"
>
  <p
    className="sort-control__text sort-control__text--sorting"
  >
    SORT BY
  </p>
  <div
    className="sort-control__select-container"
  >
    <select
      className="sort-control__select"
      data-testid="selected-sort-option"
      onChange={[Function]}
      value="RELEASE DATE"
    >
      <option
        className="sort-control__text sort-control__text--sorting-option"
        value="RELEASE DATE"
      >
        RELEASE DATE
      </option>
      <option
        className="sort-control__text sort-control__text--sorting-option"
        value="TITLE"
      >
        TITLE
      </option>
    </select>
    <span
      className="icon-arrow-down"
    />
  </div>
</div>
`;
